packages/edit-site/src/components/actions/trash-post.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { store as coreStore } from '@wordpress/core-data';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useMemo } from '@wordpress/element';
+
+export default function useMoveToTrashAction() {
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+	const { deleteEntityRecord } = useDispatch( coreStore );
+
+	return useMemo(
+		() => ( {
+			id: 'move-to-trash',
+			label: __( 'Move to Trash' ),
```

Comment:
Actions will need one more callback prop to check if they are eligible for it. A current use case we have in Templates list is some of the rows can be deleted and some not(theme's ones). That would also affect the rendering logic of the DropDownMenu.

Reply:
Oh yes, this is very basic with just the behavior of the existing action. I expect the API to evolve over time as we add actions and layouts.